contracts/Manager.sol

```diff
@@ -0,0 +1,413 @@
+            _msgSender(),
+            _tokens,
+            _amountsIn,
+            _amountsOut,
+            Basket(0)
+        );
+
+        emit NewQuantityAdjustmentProposalCreated(
+            proposalsLength, 
+            _msgSender(), 
+            _tokens,
+            _amountsIn, 
+            _amountsOut
+        );
+        return ++proposalsLength;
+    }
+
+    
+    /**
+     * Proposes a new basket defined by a list of tokens and their backing quantities. 
+     * Importantly, this type of proposal means the balances that will be required from the 
+     * proposer at the time of execution are variable. If the supply of RSV changes significantly,
+     * then much more tokens could be required to execute the proposal. 
+     * 
+     */ 
+    function proposeNewBasket(
+        address[] calldata _tokens,
+        uint256[] calldata _backing
+    )
+        external returns(uint256)
+    {
+        require(_tokens.length == _backing.length, "mismatched token quantities");
+        require(_tokens.length > 0, "no tokens in basket");
+        uint256[] memory quantitiesIn;
+        uint256[] memory quantitiesOut;
+
+        proposals[proposalsLength] = new Proposal(
+            proposalsLength,
+            _msgSender(),
+            _tokens,
+            quantitiesIn,
+            quantitiesOut,
+            new Basket(_tokens, _backing, rsvDecimals)
+        );
+
+        emit NewBasketProposalCreated(proposalsLength, _msgSender(), _tokens, _backing);
+        return ++proposalsLength;
+    }
+
+    /// Accepts a proposal for a new basket, beginning the required delay.
+    function acceptProposal(uint256 _proposalID) external onlyOperator {
+        require(proposalsLength > _proposalID, "proposals length < id");
+        proposals[_proposalID].accept(now + delay);
+        emit ProposalAccepted(_proposalID, proposals[_proposalID].proposer());
+    }
+
+    // Cancels a proposal. This can be done anytime before it is enacted by any of:
+    // 1. Proposer 2. Operator 3. Owner
+    function cancelProposal(uint256 _proposalID) external {
+        require(
+            _msgSender() == proposals[_proposalID].proposer() ||
+            _msgSender() == _owner ||
+            _msgSender() == operator, 
+            "cannot cancel"
+        );
+        proposals[_proposalID].close();
+        emit ProposalCanceled(_proposalID, proposals[_proposalID].proposer(), _msgSender());
+    }
+
+    /// Executes a proposal by exchanging collateral tokens with the proposer.
+    function executeProposal(uint256 _proposalID) external {
+        require(proposalsLength > _proposalID, "proposals length < id");
+        (address[] memory tokens, uint256[] memory quantitiesIn, uint256[] memory quantitiesOut) =
+            proposals[_proposalID].complete(rsv.totalSupply(), address(vault), basket);
+
+        // Proposer -> Vault
+        for (uint i = 0; i < tokens.length; i++) {
+            IERC20(tokens[i]).safeTransferFrom(
```

Comment:
`if (quantitiesIn[i] > 0) {` ?

I realize that this would add to the bytecode length. Might be fine to remove, but seems a bit unfortunate if so. Should at least make sure that the backing tokens all allow transfers of size zero.